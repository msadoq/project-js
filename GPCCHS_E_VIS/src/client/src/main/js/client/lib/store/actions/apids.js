import simple from 'store/helpers/simpleActionCreator';
import {
  WS_APIDS_ASK,
  WS_APIDS_ADD,
} from '../types';

export const askApids = simple(
  WS_APIDS_ASK,
  'domainId',
  'sessionId'
);

export const addApids = simple(
  WS_APIDS_ADD,
  'tupleId',
  'apids'
);
