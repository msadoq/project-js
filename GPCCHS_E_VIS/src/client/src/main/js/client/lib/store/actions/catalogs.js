import simple from 'store/helpers/simpleActionCreator';
import {
  WS_CATALOGS_ASK,
  WS_CATALOGS_ADD,
  WS_CATALOG_ITEMS_ASK,
  WS_CATALOG_ITEMS_ADD,
  WS_COM_OBJECTS_ASK,
  WS_COM_OBJECTS_ADD,
  WS_UNIT_ASK,
  WS_UNIT_ADD,
  WS_ASK_CATALOGS_AND_ITEMS_AND_UNIT,
} from '../types';

export const askCatalogs = simple(
  WS_CATALOGS_ASK,
  'domainId',
  'sessionId'
);

export const addCatalogs = simple(
  WS_CATALOGS_ADD,
  'tupleId',
  'catalogs'
);

export const askCatalogItems = simple(
  WS_CATALOG_ITEMS_ASK,
  'domainId',
  'sessionId',
  'name'
);

export const addCatalogItems = simple(
  WS_CATALOG_ITEMS_ADD,
  'tupleId',
  'name',
  'items'
);

export const askComObjects = simple(
  WS_COM_OBJECTS_ASK,
  'domainId',
  'sessionId',
  'name',
  'itemName'
);

export const addComObjects = simple(
  WS_COM_OBJECTS_ADD,
  'tupleId',
  'name',
  'itemName',
  'comObjects'
);

export const askUnit = simple(
  WS_UNIT_ASK,
  'domainId',
  'sessionId',
  'name',
  'itemName'
);

export const addUnit = simple(
  WS_UNIT_ADD,
  'tupleId',
  'name',
  'itemName',
  'unit'
);

export const askCatalogsAndItemsAndUnit = simple(
  WS_ASK_CATALOGS_AND_ITEMS_AND_UNIT,
  'domainId',
  'sessionId',
  'name',
  'itemName'
);
