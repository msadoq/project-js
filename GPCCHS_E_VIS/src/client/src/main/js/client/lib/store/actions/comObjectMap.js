import simple from 'store/helpers/simpleActionCreator';
import {
  SET_COM_OBJECT_MAP,
} from '../types';

const setComObjectMap = simple(
  SET_COM_OBJECT_MAP,
  'newComObjectMap'
);

export default setComObjectMap;
