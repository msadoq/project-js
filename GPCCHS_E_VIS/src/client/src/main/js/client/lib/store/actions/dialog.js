import simple from '../helpers/simpleActionCreator';
import * as types from '../types';

const openDialog = simple(types.HSC_OPEN_DIALOG, 'windowId', 'id', 'type', 'options');

const closeDialog = simple(
  types.HSC_CLOSE_DIALOG,
  'windowId',
  'id',
  'choice'
);

export default {
  openDialog,
  closeDialog,
};
