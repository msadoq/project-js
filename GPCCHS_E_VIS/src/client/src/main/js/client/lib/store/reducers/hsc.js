import _omit from 'lodash/omit';
import * as types from '../types';

const initialState = {
  windowsOpened: false,
  workspaceOpened: false,
  playingTimebarId: null,
  lastCacheInvalidation: Date.now(),
  folder: null,
  file: null,
  focusWindow: null,
};

export default function hsc(state = initialState, action) {
  switch (action.type) {
    case types.HSC_PLAY:
      return Object.assign({}, state, { playingTimebarId: action.payload.timebarUuid });
    case types.HSC_PAUSE:
      return Object.assign({}, state, { playingTimebarId: null });
    case types.HSC_SET_WINDOWS_AS_OPENED:
      return Object.assign({}, state, { windowsOpened: true });
    case types.HSC_UPDATE_LAST_CACHE_INVALIDATION:
      return Object.assign({}, state, { lastCacheInvalidation: action.payload.timestamp });
    case types.HSC_ISWORKSPACE_OPENING:
      return Object.assign({}, state, { isWorkspaceOpening: action.payload.flag });
    case types.HSC_UPDATE_PATH:
      return Object.assign({}, state, { folder: action.payload.folder, file: action.payload.file });
    case types.HSC_CLOSE_WORKSPACE:
      return _omit(state, ['folder', 'file']);
    case types.HSC_FOCUS_WINDOW:
      return Object.assign({}, state, {
        focusWindow: action.payload.windowId,
      });
    case types.HSC_BLUR_WINDOW:
      if (state.focusWindow === action.payload.windowId) {
        return Object.assign({}, state, {
          focusWindow: null,
        });
      }
      return state;
    default:
      return state;
  }
}
