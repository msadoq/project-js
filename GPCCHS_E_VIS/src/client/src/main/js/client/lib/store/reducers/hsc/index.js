import _ from 'lodash/fp';
import _omit from 'lodash/omit';
import * as types from '../../types';

/* --- Reducer -------------------------------------------------------------- */

const initialState = {
  windowsOpened: false,
  playingTimebarId: null,
  lastCacheInvalidation: Date.now(),
  folder: null,
  file: null,
  focusWindow: null,
  isModified: true,
  forecast: {},
  domainName: null,
  sessionName: null,
};

/* eslint-disable complexity, "DV6 TBC_CNES Redux reducers should be implemented as switch case" */
export default function hsc(state = initialState, action) {
  switch (action.type) {
    case types.HSC_PLAY:
      return Object.assign({}, state, { playingTimebarId: action.payload.timebarUuid });
    case types.HSC_PAUSE:
      return Object.assign({}, state, { playingTimebarId: null });
    case types.HSC_SET_WINDOWS_AS_OPENED:
      return Object.assign({}, state, { windowsOpened: true });
    case types.HSC_UPDATE_LAST_CACHE_INVALIDATION:
      return Object.assign({}, state, { lastCacheInvalidation: action.payload.timestamp });
    case types.HSC_ISWORKSPACE_OPENING:
      return Object.assign({}, state, { isWorkspaceOpening: action.payload.flag });
    case types.HSC_UPDATE_PATH:
      return Object.assign({}, state, { folder: action.payload.folder, file: action.payload.file });
    case types.HSC_CLOSE_WORKSPACE:
      return _omit(state, ['folder', 'file']);
    case types.HSC_FOCUS_WINDOW:
      return Object.assign({}, state, {
        focusWindow: action.payload.windowId,
      });
    case types.HSC_BLUR_WINDOW:
      if (state.focusWindow === action.payload.windowId) {
        return Object.assign({}, state, {
          focusWindow: null,
        });
      }
      return state;
    case types.WS_TIMEBAR_UPDATE_CURSORS:
    case types.WS_TIMELINE_CREATE_NEW:
    case types.WS_TIMELINE_REMOVE:
    case types.WS_PAGE_UPDATE_TIMEBARID:
    case types.WS_WINDOW_PAGE_REORDER:
    case types.WS_WINDOW_MOVE_TAB_ORDER:
    case types.WS_PAGE_UPDATE_ABSOLUTEPATH:
    case types.WS_WINDOW_UPDATE_TITLE:
    case types.WS_WINDOW_UPDATE_DOMAINNAME:
    case types.WS_WINDOW_UPDATE_SESSIONNAME:
      return _.set('isModified', true, state);
    case types.WS_WORKSPACE_SET_MODIFIED:
      return _.set('isModified', action.payload.flag, state);
    case types.WS_WORKSPACE_OPEN:
      return _.set('isModified', false, state);
    // Forecast Management
    case types.WS_PAGE_OPEN:
    case types.WS_PAGE_ADD_BLANK:
    case types.WS_PAGE_CLOSE:
      return Object.assign({}, state, {
        forecast: {},
        isModified: true,
      });
    case types.WS_WINDOW_PAGE_FOCUS:
      return Object.assign({}, state, { forecast: {} });
    case types.HSC_UPDATE_FORECAST:
      return Object.assign({}, state, { forecast: {
        upper: action.payload.upper,
        lower: action.payload.lower,
      } });
    case types.WS_WORKSPACE_UPDATE_DOMAINNAME:
      if (action.payload.domainName) {
        return { ...state, domainName: action.payload.domainName, isModified: true };
      }
      return Object.assign({}, _.omit('domainName', state), { isModified: true });
    case types.WS_WORKSPACE_UPDATE_SESSIONNAME:
      if (action.payload.sessionName) {
        return { ...state, sessionName: action.payload.sessionName, isModified: true };
      }
      return Object.assign({}, _.omit('sessionName', state), { isModified: true });
    default:
      return state;
  }
}

/* --- Selectors ------------------------------------------------------------ */

const inHsc = key => _.path(['hsc', key]);

// simples
export const getWorkspaceFile = inHsc('file');
export const getWorkspaceFolder = inHsc('folder');
export const getWindowsOpened = inHsc('windowsOpened');
export const getLastCacheInvalidation = inHsc('lastCacheInvalidation');
export const getPlayingTimebarId = inHsc('playingTimebarId');
export const getFocusedWindowId = inHsc('focusWindow');
export const getIsWorkspaceOpening = inHsc('isWorkspaceOpening');
export const getWorkspaceIsModified = inHsc('isModified');
export const getForecast = inHsc('forecast');
export const getDomainName = inHsc('domainName');
export const getSessionName = inHsc('sessionName');
