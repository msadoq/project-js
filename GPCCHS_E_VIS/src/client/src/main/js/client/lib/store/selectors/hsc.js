import { getWindowPageIds } from '../reducers/windows';
import { getPage } from '../reducers/pages';
import { getView } from '../reducers/views';

/**
 * Return true if at least one document opened in workspace should be saved
 *
 * @param state
 * @param windowId
 * @returns {boolean}
 */
const getIsSaveNeeded = (state, { windowId }) => {
  const pageIds = getWindowPageIds(state, { windowId }) || [];
  // is there is a page to save?
  let viewIds = [];
  const pageToSave = pageIds.find((pageId) => {
    const page = getPage(state, { pageId });
    if (page && page.views && page.views.length) {
      viewIds = viewIds.concat(page.views);
    }
    return page && !!page.isModified;
  });
  if (typeof pageToSave !== 'undefined') {
    return true;
  }

  // is there is a view to save?
  const viewToSave = viewIds.find((viewId) => {
    const view = getView(state, { viewId });
    return view && !!view.isModified;
  });
  if (typeof viewToSave !== 'undefined') {
    return true;
  }

  // nothing to save in this window
  return false;
};

export default {
  getIsSaveNeeded,
};
