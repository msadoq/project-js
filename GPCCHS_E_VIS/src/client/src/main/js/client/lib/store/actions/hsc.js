import { HEALTH_STATUS_CRITICAL } from 'common/constants';
import _keys from 'lodash/keys';
import simple from '../simpleActionCreator';
import * as types from '../types';
import { getHealthMap } from '../selectors/health';
import { getTimebars } from '../selectors/timebars';
import { setRealTime } from './timebars';
import { addOnce } from './messages';

/**
 * App lifecycle
 */
export const setWindowsAsOpened = simple(types.HSC_SET_WINDOWS_AS_OPENED);
export const isWorkspaceOpening = simple(types.HSC_ISWORKSPACE_OPENING, 'flag');
export const closeWorkspace = simple(types.HSC_CLOSE_WORKSPACE);

/**
 * Play mode
 */
export const play = simple(types.HSC_PLAY, 'timebarUuid');
export const smartPlay = timebarUuid => // TODO dbrugne test
  (dispatch, getState) => {
    const health = getHealthMap(getState());
    if (
      health.dc !== HEALTH_STATUS_CRITICAL
      && health.hss !== HEALTH_STATUS_CRITICAL
      && health.main !== HEALTH_STATUS_CRITICAL
      && health.windows !== HEALTH_STATUS_CRITICAL
    ) {
      dispatch(play(timebarUuid));
    } else {
      dispatch(addOnce(
        'global',
        'warning',
        'One process of the application is oveloaded, cannot switch to play'
        )
      );
    }
  };
export const pause = () =>
  (dispatch, getState) => {
    dispatch({ type: types.HSC_PAUSE });
    _keys(getTimebars(getState())).forEach((timebarId) => {
      dispatch(setRealTime(timebarId, false));
    });
  };

/**
 * Cache invalidation
 */
export const updateCacheInvalidation
  = simple(types.HSC_UPDATE_LAST_CACHE_INVALIDATION, 'timestamp');

/**
 * workspace path
 */
export const updatePath = simple(types.HSC_UPDATE_PATH, 'folder', 'file');


/**
 * Save focus/blurred window
 */
export const focusWindow = simple(types.HSC_FOCUS_WINDOW, 'windowId');
export const blurWindow = simple(types.HSC_BLUR_WINDOW, 'windowId');
