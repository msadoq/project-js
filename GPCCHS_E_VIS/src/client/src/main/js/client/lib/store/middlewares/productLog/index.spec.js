import configureMockStore from 'redux-mock-store';
import { sendProductLog } from '../../actions/hsc';
import makeProductLogMiddleware from './';

const fakeSendProductLog = x => x;
const mockStore = configureMockStore([makeProductLogMiddleware(fakeSendProductLog)]);

describe('mainProcess:store:middlewares:wikiHelper', () => {
  const store = mockStore({});

  beforeEach(() => {
  });

  afterEach(() => {
    store.clearActions();
  });

  test('propagate unknown action', () => {
    store.dispatch({ type: 'DUMMY_ACTION' });
    expect(store.getActions()).toEqual([{ type: 'DUMMY_ACTION' }]);
  });

  test('send product logs', () => {
    store.dispatch(sendProductLog(1337, '1', '2', '3'));
    expect(store.getActions()).toMatchSnapshot();
  });
});
