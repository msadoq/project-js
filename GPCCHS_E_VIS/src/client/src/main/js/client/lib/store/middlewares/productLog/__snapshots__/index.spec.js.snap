// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mainProcess:store:middlewares:wikiHelper send product logs 1`] = `
Array [
  Object {
    "payload": Object {
      "args": Array [
        "1",
        "2",
        "3",
      ],
      "uid": 1337,
    },
    "type": "HSC_SEND_PRODUCT_LOG",
  },
]
`;
